java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPAddress.java

```diff
@@ -169,4 +183,26 @@ else if (address instanceof Inet6Address)
     public static IPAddress<?> fromInetAddress(@Nonnull InetAddress address) {
         return of(address);
     }
+
+    @Nonnull
+    protected <T extends InetAddress> T toInetAddressSubclassChecked(
```

Comment:
not sure that this protected method buys us anything. IMHO it would be better/clearer to remove it and just have

```java
Inet4Address toInetAddress() {
    try {
      return (Inet4Address) InetADdress.getByAddress(getBytes());
    } catch (UnknownHostException e) {
      // ...
    }
}
```

in the subclasses directly. 

I feel this adds quite a bit more structural complexity than it removes from those methods...